tests/callbacks/test_gpu_stats_monitor.py

```diff
@@ -116,6 +116,40 @@ def test_gpu_stats_monitor_no_gpu_warning(tmpdir):
 
 
 def test_gpu_stats_monitor_parse_gpu_stats():
-    logs = GPUStatsMonitor._parse_gpu_stats('1,2', [[3, 4, 5], [6, 7]], [('gpu', 'a'), ('memory', 'b')])
-    expected = {'gpu_id: 1/gpu (a)': 3, 'gpu_id: 1/memory (b)': 4, 'gpu_id: 2/gpu (a)': 6, 'gpu_id: 2/memory (b)': 7}
+    logs = GPUStatsMonitor._parse_gpu_stats([1, 2], [[3, 4, 5], [6, 7]], [('gpu', 'a'), ('memory', 'b')])
+    expected = {
+        'device_id: 1/gpu (a)': 3,
+        'device_id: 1/memory (b)': 4,
+        'device_id: 2/gpu (a)': 6,
+        'device_id: 2/memory (b)': 7
+    }
     assert logs == expected
+
+
+@RunIf(min_gpus=2)
+def test_gpu_stats_monitor_get_gpu_ids_cuda_visible_devices_unset():
+    # `CUDA_VISIBLE_DEVICES` unset
+    try:
+        del os.environ['CUDA_VISIBLE_DEVICES']
+    except KeyError:
+        pass
+
+    gpu_ids = GPUStatsMonitor._get_gpu_ids([1, 0])
+    expected = ['1', '0']
+    assert gpu_ids == expected
```

Comment:
The `CUDA_VISIBLE_DEVICES` should always been unset before the test program starting. The manipulation with `os.environ['CUDA_VISIBLE_DEVICES']` will not work after `torch.cuda` is imported.

```python
$ unset CUDA_VISIBLE_DEIVCES
$ ipython3
In [1]: import torch

In [2]: torch.cuda.device_count()
Out[2]: 10

In [3]: import os

In [4]: os.environ['CUDA_VISIBLE_DEVICES'] = '0'

In [5]: torch.cuda.device_count()
Out[5]: 10
```

```python
$ unset CUDA_VISIBLE_DEIVCES
$ ipython3
In [1]: import os

In [2]: os.environ['CUDA_VISIBLE_DEVICES'] = '0'

In [1]: import torch

In [3]: torch.cuda.device_count()
Out[3]: 1
```